File: sdkwork-backend-sdk-go/api/channel.go
package api

import (
    "fmt"
    sdktypes "github.com/sdkwork/backend-sdk/types"
    sdkhttp "github.com/sdkwork/backend-sdk/http"
)

type ChannelApi struct {
    client *sdkhttp.Client
}

func NewChannelApi(client *sdkhttp.Client) *ChannelApi {
    return &ChannelApi{client: client}
}

// Update channel
func (a *ChannelApi) Update(body sdktypes.PlusChannelForm) (sdktypes.PlusApiResultPlusChannelVO, error) {
    raw, err := a.client.Put(BackendApiPath("/channel"), body, nil, nil, "")
    if err != nil {
        var zero sdktypes.PlusApiResultPlusChannelVO
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultPlusChannelVO](raw)
}

// Create channel
func (a *ChannelApi) Create(body sdktypes.PlusChannelForm) (sdktypes.PlusApiResultPlusChannelVO, error) {
    raw, err := a.client.Post(BackendApiPath("/channel"), body, nil, nil, "")
    if err != nil {
        var zero sdktypes.PlusApiResultPlusChannelVO
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultPlusChannelVO](raw)
}

// Update an existing channel resource
func (a *ChannelApi) UpdateResource(body sdktypes.PlusChannelResourceForm) (sdktypes.PlusApiResultPlusChannelResourceVO, error) {
    raw, err := a.client.Put(BackendApiPath("/channel/resource"), body, nil, nil, "")
    if err != nil {
        var zero sdktypes.PlusApiResultPlusChannelResourceVO
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultPlusChannelResourceVO](raw)
}

// Create a new channel resource
func (a *ChannelApi) CreateResource(body sdktypes.PlusChannelResourceForm) (sdktypes.PlusApiResultPlusChannelResourceVO, error) {
    raw, err := a.client.Post(BackendApiPath("/channel/resource"), body, nil, nil, "")
    if err != nil {
        var zero sdktypes.PlusApiResultPlusChannelResourceVO
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultPlusChannelResourceVO](raw)
}

// Update channel proxy
func (a *ChannelApi) UpdateProxy(body sdktypes.PlusChannelProxyForm) (sdktypes.PlusApiResultPlusChannelProxyVO, error) {
    raw, err := a.client.Put(BackendApiPath("/channel/proxy"), body, nil, nil, "")
    if err != nil {
        var zero sdktypes.PlusApiResultPlusChannelProxyVO
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultPlusChannelProxyVO](raw)
}

// Create channel proxy
func (a *ChannelApi) CreateProxy(body sdktypes.PlusChannelProxyForm) (sdktypes.PlusApiResultPlusChannelProxyVO, error) {
    raw, err := a.client.Post(BackendApiPath("/channel/proxy"), body, nil, nil, "")
    if err != nil {
        var zero sdktypes.PlusApiResultPlusChannelProxyVO
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultPlusChannelProxyVO](raw)
}

// Update channel account
func (a *ChannelApi) UpdateAccount(body sdktypes.PlusChannelAccountForm) (sdktypes.PlusApiResultPlusChannelAccountVO, error) {
    raw, err := a.client.Put(BackendApiPath("/channel/account"), body, nil, nil, "")
    if err != nil {
        var zero sdktypes.PlusApiResultPlusChannelAccountVO
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultPlusChannelAccountVO](raw)
}

// Create channel account
func (a *ChannelApi) CreateAccount(body sdktypes.PlusChannelAccountForm) (sdktypes.PlusApiResultPlusChannelAccountVO, error) {
    raw, err := a.client.Post(BackendApiPath("/channel/account"), body, nil, nil, "")
    if err != nil {
        var zero sdktypes.PlusApiResultPlusChannelAccountVO
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultPlusChannelAccountVO](raw)
}

// Get channel resources by page
func (a *ChannelApi) CreateListByPage(body *sdktypes.QueryListForm, query map[string]interface{}) (sdktypes.PlusApiResultPagePlusChannelResourceVO, error) {
    raw, err := a.client.Post(BackendApiPath("/channel/resource/list"), body, query, nil, "")
    if err != nil {
        var zero sdktypes.PlusApiResultPagePlusChannelResourceVO
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultPagePlusChannelResourceVO](raw)
}

// Get all channel resources
func (a *ChannelApi) CreateListAllEntities(body *sdktypes.QueryListForm) (sdktypes.PlusApiResultListPlusChannelResourceVO, error) {
    raw, err := a.client.Post(BackendApiPath("/channel/resource/list/all"), body, nil, nil, "")
    if err != nil {
        var zero sdktypes.PlusApiResultListPlusChannelResourceVO
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultListPlusChannelResourceVO](raw)
}

// List channel proxies by page
func (a *ChannelApi) CreateListByPageProxy(body *sdktypes.QueryListForm, query map[string]interface{}) (sdktypes.PlusApiResultPagePlusChannelProxyVO, error) {
    raw, err := a.client.Post(BackendApiPath("/channel/proxy/list"), body, query, nil, "")
    if err != nil {
        var zero sdktypes.PlusApiResultPagePlusChannelProxyVO
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultPagePlusChannelProxyVO](raw)
}

// List all channel proxies
func (a *ChannelApi) CreateListAllEntitiesProxy(body *sdktypes.QueryListForm) (sdktypes.PlusApiResultListPlusChannelProxyVO, error) {
    raw, err := a.client.Post(BackendApiPath("/channel/proxy/list/all"), body, nil, nil, "")
    if err != nil {
        var zero sdktypes.PlusApiResultListPlusChannelProxyVO
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultListPlusChannelProxyVO](raw)
}

// List channels by page
func (a *ChannelApi) CreateListByPageChannel(body *sdktypes.QueryListForm, query map[string]interface{}) (sdktypes.PlusApiResultPagePlusChannelVO, error) {
    raw, err := a.client.Post(BackendApiPath("/channel/list"), body, query, nil, "")
    if err != nil {
        var zero sdktypes.PlusApiResultPagePlusChannelVO
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultPagePlusChannelVO](raw)
}

// List all channels
func (a *ChannelApi) CreateListAllEntitiesChannel(body *sdktypes.QueryListForm) (sdktypes.PlusApiResultListPlusChannelVO, error) {
    raw, err := a.client.Post(BackendApiPath("/channel/list/all"), body, nil, nil, "")
    if err != nil {
        var zero sdktypes.PlusApiResultListPlusChannelVO
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultListPlusChannelVO](raw)
}

// Get SMS channel accounts
func (a *ChannelApi) ListSms(body *sdktypes.PlusChannelAccountQueryForm) (sdktypes.PlusApiResultListPlusChannelAccountVO, error) {
    raw, err := a.client.Post(BackendApiPath("/channel/account/list_sms"), body, nil, nil, "")
    if err != nil {
        var zero sdktypes.PlusApiResultListPlusChannelAccountVO
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultListPlusChannelAccountVO](raw)
}

// Get payment channel accounts
func (a *ChannelApi) ListPayment(body *sdktypes.PlusChannelAccountQueryForm) (sdktypes.PlusApiResultListPlusChannelAccountVO, error) {
    raw, err := a.client.Post(BackendApiPath("/channel/account/list_payment"), body, nil, nil, "")
    if err != nil {
        var zero sdktypes.PlusApiResultListPlusChannelAccountVO
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultListPlusChannelAccountVO](raw)
}

// Get OSS channel accounts
func (a *ChannelApi) ListOss(body *sdktypes.PlusChannelAccountQueryForm) (sdktypes.PlusApiResultListPlusChannelAccountVO, error) {
    raw, err := a.client.Post(BackendApiPath("/channel/account/list_oss"), body, nil, nil, "")
    if err != nil {
        var zero sdktypes.PlusApiResultListPlusChannelAccountVO
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultListPlusChannelAccountVO](raw)
}

// Get official account channel accounts
func (a *ChannelApi) ListOfficialAccount(body *sdktypes.PlusChannelAccountQueryForm) (sdktypes.PlusApiResultListPlusChannelAccountVO, error) {
    raw, err := a.client.Post(BackendApiPath("/channel/account/list_official_account"), body, nil, nil, "")
    if err != nil {
        var zero sdktypes.PlusApiResultListPlusChannelAccountVO
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultListPlusChannelAccountVO](raw)
}

// Get mini program channel accounts
func (a *ChannelApi) ListMiniProgram(body *sdktypes.PlusChannelAccountQueryForm) (sdktypes.PlusApiResultListPlusChannelAccountVO, error) {
    raw, err := a.client.Post(BackendApiPath("/channel/account/list_mini_program"), body, nil, nil, "")
    if err != nil {
        var zero sdktypes.PlusApiResultListPlusChannelAccountVO
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultListPlusChannelAccountVO](raw)
}

// Get channel accounts by page
func (a *ChannelApi) CreateListByPageAccount(body *sdktypes.QueryListForm, query map[string]interface{}) (sdktypes.PlusApiResultPagePlusChannelAccountVO, error) {
    raw, err := a.client.Post(BackendApiPath("/channel/account/list"), body, query, nil, "")
    if err != nil {
        var zero sdktypes.PlusApiResultPagePlusChannelAccountVO
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultPagePlusChannelAccountVO](raw)
}

// Get all channel accounts
func (a *ChannelApi) CreateListAllEntitiesAccount(body *sdktypes.QueryListForm) (sdktypes.PlusApiResultListPlusChannelAccountVO, error) {
    raw, err := a.client.Post(BackendApiPath("/channel/account/list/all"), body, nil, nil, "")
    if err != nil {
        var zero sdktypes.PlusApiResultListPlusChannelAccountVO
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultListPlusChannelAccountVO](raw)
}

// Get channel by ID
func (a *ChannelApi) GetById(id string) (sdktypes.PlusApiResultPlusChannelVO, error) {
    raw, err := a.client.Get(BackendApiPath(fmt.Sprintf("/channel/%s", id)), nil, nil)
    if err != nil {
        var zero sdktypes.PlusApiResultPlusChannelVO
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultPlusChannelVO](raw)
}

// Delete channel
func (a *ChannelApi) Delete(id string) (sdktypes.PlusApiResultBoolean, error) {
    raw, err := a.client.Delete(BackendApiPath(fmt.Sprintf("/channel/%s", id)), nil, nil)
    if err != nil {
        var zero sdktypes.PlusApiResultBoolean
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultBoolean](raw)
}

// Get a channel resource by ID
func (a *ChannelApi) GetByIdResource(id string) (sdktypes.PlusApiResultPlusChannelResourceVO, error) {
    raw, err := a.client.Get(BackendApiPath(fmt.Sprintf("/channel/resource/%s", id)), nil, nil)
    if err != nil {
        var zero sdktypes.PlusApiResultPlusChannelResourceVO
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultPlusChannelResourceVO](raw)
}

// Delete a channel resource
func (a *ChannelApi) DeleteResource(id string) (sdktypes.PlusApiResultBoolean, error) {
    raw, err := a.client.Delete(BackendApiPath(fmt.Sprintf("/channel/resource/%s", id)), nil, nil)
    if err != nil {
        var zero sdktypes.PlusApiResultBoolean
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultBoolean](raw)
}

// Get channel proxy by ID
func (a *ChannelApi) GetByIdProxy(id string) (sdktypes.PlusApiResultPlusChannelProxyVO, error) {
    raw, err := a.client.Get(BackendApiPath(fmt.Sprintf("/channel/proxy/%s", id)), nil, nil)
    if err != nil {
        var zero sdktypes.PlusApiResultPlusChannelProxyVO
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultPlusChannelProxyVO](raw)
}

// Delete channel proxy
func (a *ChannelApi) DeleteProxy(id string) (sdktypes.PlusApiResultBoolean, error) {
    raw, err := a.client.Delete(BackendApiPath(fmt.Sprintf("/channel/proxy/%s", id)), nil, nil)
    if err != nil {
        var zero sdktypes.PlusApiResultBoolean
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultBoolean](raw)
}

// Get channel account by ID
func (a *ChannelApi) GetByIdAccount(id string) (sdktypes.PlusApiResultPlusChannelAccountVO, error) {
    raw, err := a.client.Get(BackendApiPath(fmt.Sprintf("/channel/account/%s", id)), nil, nil)
    if err != nil {
        var zero sdktypes.PlusApiResultPlusChannelAccountVO
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultPlusChannelAccountVO](raw)
}

// Delete channel account
func (a *ChannelApi) DeleteAccount(id string) (sdktypes.PlusApiResultBoolean, error) {
    raw, err := a.client.Delete(BackendApiPath(fmt.Sprintf("/channel/account/%s", id)), nil, nil)
    if err != nil {
        var zero sdktypes.PlusApiResultBoolean
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultBoolean](raw)
}

File: sdkwork-backend-sdk-typescript/src/api/channel.ts
import { backendApiPath } from './paths';
import type { HttpClient } from '../http/client';
import type { QueryParams } from '../types/common';
import type { PlusApiResultBoolean, PlusApiResultListPlusChannelAccountVO, PlusApiResultListPlusChannelProxyVO, PlusApiResultListPlusChannelResourceVO, PlusApiResultListPlusChannelVO, PlusApiResultPagePlusChannelAccountVO, PlusApiResultPagePlusChannelProxyVO, PlusApiResultPagePlusChannelResourceVO, PlusApiResultPagePlusChannelVO, PlusApiResultPlusChannelAccountVO, PlusApiResultPlusChannelProxyVO, PlusApiResultPlusChannelResourceVO, PlusApiResultPlusChannelVO, PlusApiResultWechatConnectivityCheckResponse, PlusApiResultWechatDomainVerifyResponse, PlusChannelAccountForm, PlusChannelAccountQueryForm, PlusChannelForm, PlusChannelProxyForm, PlusChannelResourceForm, QueryListForm, WechatConnectivityCheckRequest, WechatDomainVerifyRequest } from '../types';


export class ChannelApi {
  private client: HttpClient;
  
  constructor(client: HttpClient) { 
    this.client = client; 
  }

/** Update channel */
  async update(body: PlusChannelForm): Promise<PlusApiResultPlusChannelVO> {
    return this.client.put<PlusApiResultPlusChannelVO>(backendApiPath(`/channel`), body);
  }

/** Create channel */
  async create(body: PlusChannelForm): Promise<PlusApiResultPlusChannelVO> {
    return this.client.post<PlusApiResultPlusChannelVO>(backendApiPath(`/channel`), body);
  }

/** Update an existing channel resource */
  async updateResource(body: PlusChannelResourceForm): Promise<PlusApiResultPlusChannelResourceVO> {
    return this.client.put<PlusApiResultPlusChannelResourceVO>(backendApiPath(`/channel/resource`), body);
  }

/** Create a new channel resource */
  async createResource(body: PlusChannelResourceForm): Promise<PlusApiResultPlusChannelResourceVO> {
    return this.client.post<PlusApiResultPlusChannelResourceVO>(backendApiPath(`/channel/resource`), body);
  }

/** Update channel proxy */
  async updateProxy(body: PlusChannelProxyForm): Promise<PlusApiResultPlusChannelProxyVO> {
    return this.client.put<PlusApiResultPlusChannelProxyVO>(backendApiPath(`/channel/proxy`), body);
  }

/** Create channel proxy */
  async createProxy(body: PlusChannelProxyForm): Promise<PlusApiResultPlusChannelProxyVO> {
    return this.client.post<PlusApiResultPlusChannelProxyVO>(backendApiPath(`/channel/proxy`), body);
  }

/** Update channel account */
  async updateAccount(body: PlusChannelAccountForm): Promise<PlusApiResultPlusChannelAccountVO> {
    return this.client.put<PlusApiResultPlusChannelAccountVO>(backendApiPath(`/channel/account`), body);
  }

/** Create channel account */
  async createAccount(body: PlusChannelAccountForm): Promise<PlusApiResultPlusChannelAccountVO> {
    return this.client.post<PlusApiResultPlusChannelAccountVO>(backendApiPath(`/channel/account`), body);
  }

/** Get channel resources by page */
  async createListByPage(body?: QueryListForm, params?: QueryParams): Promise<PlusApiResultPagePlusChannelResourceVO> {
    return this.client.post<PlusApiResultPagePlusChannelResourceVO>(backendApiPath(`/channel/resource/list`), body, params);
  }

/** Get all channel resources */
  async createListAllEntities(body?: QueryListForm): Promise<PlusApiResultListPlusChannelResourceVO> {
    return this.client.post<PlusApiResultListPlusChannelResourceVO>(backendApiPath(`/channel/resource/list/all`), body);
  }

/** List channel proxies by page */
  async createListByPageProxy(body?: QueryListForm, params?: QueryParams): Promise<PlusApiResultPagePlusChannelProxyVO> {
    return this.client.post<PlusApiResultPagePlusChannelProxyVO>(backendApiPath(`/channel/proxy/list`), body, params);
  }

/** List all channel proxies */
  async createListAllEntitiesProxy(body?: QueryListForm): Promise<PlusApiResultListPlusChannelProxyVO> {
    return this.client.post<PlusApiResultListPlusChannelProxyVO>(backendApiPath(`/channel/proxy/list/all`), body);
  }

/** List channels by page */
  async createListByPageChannel(body?: QueryListForm, params?: QueryParams): Promise<PlusApiResultPagePlusChannelVO> {
    return this.client.post<PlusApiResultPagePlusChannelVO>(backendApiPath(`/channel/list`), body, params);
  }

/** List all channels */
  async createListAllEntitiesChannel(body?: QueryListForm): Promise<PlusApiResultListPlusChannelVO> {
    return this.client.post<PlusApiResultListPlusChannelVO>(backendApiPath(`/channel/list/all`), body);
  }

/** Verify WeChat official account domains */
  async verifyWechatDomains(id: string | number, body?: WechatDomainVerifyRequest): Promise<PlusApiResultWechatDomainVerifyResponse> {
    return this.client.post<PlusApiResultWechatDomainVerifyResponse>(backendApiPath(`/channel/account/${id}/wechat/domain/verify`), body);
  }

/** Check WeChat official account connectivity */
  async checkWechatConnectivity(id: string | number, body?: WechatConnectivityCheckRequest): Promise<PlusApiResultWechatConnectivityCheckResponse> {
    return this.client.post<PlusApiResultWechatConnectivityCheckResponse>(backendApiPath(`/channel/account/${id}/wechat/connectivity/check`), body);
  }

/** Get SMS channel accounts */
  async listSms(body?: PlusChannelAccountQueryForm): Promise<PlusApiResultListPlusChannelAccountVO> {
    return this.client.post<PlusApiResultListPlusChannelAccountVO>(backendApiPath(`/channel/account/list_sms`), body);
  }

/** Get payment channel accounts */
  async listPayment(body?: PlusChannelAccountQueryForm): Promise<PlusApiResultListPlusChannelAccountVO> {
    return this.client.post<PlusApiResultListPlusChannelAccountVO>(backendApiPath(`/channel/account/list_payment`), body);
  }

/** Get OSS channel accounts */
  async listOss(body?: PlusChannelAccountQueryForm): Promise<PlusApiResultListPlusChannelAccountVO> {
    return this.client.post<PlusApiResultListPlusChannelAccountVO>(backendApiPath(`/channel/account/list_oss`), body);
  }

/** Get official account channel accounts */
  async listOfficialAccount(body?: PlusChannelAccountQueryForm): Promise<PlusApiResultListPlusChannelAccountVO> {
    return this.client.post<PlusApiResultListPlusChannelAccountVO>(backendApiPath(`/channel/account/list_official_account`), body);
  }

/** Get mini program channel accounts */
  async listMiniProgram(body?: PlusChannelAccountQueryForm): Promise<PlusApiResultListPlusChannelAccountVO> {
    return this.client.post<PlusApiResultListPlusChannelAccountVO>(backendApiPath(`/channel/account/list_mini_program`), body);
  }

/** Get channel accounts by page */
  async createListByPageAccount(body?: QueryListForm, params?: QueryParams): Promise<PlusApiResultPagePlusChannelAccountVO> {
    return this.client.post<PlusApiResultPagePlusChannelAccountVO>(backendApiPath(`/channel/account/list`), body, params);
  }

/** Get all channel accounts */
  async createListAllEntitiesAccount(body?: QueryListForm): Promise<PlusApiResultListPlusChannelAccountVO> {
    return this.client.post<PlusApiResultListPlusChannelAccountVO>(backendApiPath(`/channel/account/list/all`), body);
  }

/** Get channel by ID */
  async getById(id: string | number): Promise<PlusApiResultPlusChannelVO> {
    return this.client.get<PlusApiResultPlusChannelVO>(backendApiPath(`/channel/${id}`));
  }

/** Delete channel */
  async delete(id: string | number): Promise<PlusApiResultBoolean> {
    return this.client.delete<PlusApiResultBoolean>(backendApiPath(`/channel/${id}`));
  }

/** Get a channel resource by ID */
  async getByIdResource(id: string | number): Promise<PlusApiResultPlusChannelResourceVO> {
    return this.client.get<PlusApiResultPlusChannelResourceVO>(backendApiPath(`/channel/resource/${id}`));
  }

/** Delete a channel resource */
  async deleteResource(id: string | number): Promise<PlusApiResultBoolean> {
    return this.client.delete<PlusApiResultBoolean>(backendApiPath(`/channel/resource/${id}`));
  }

/** Get channel proxy by ID */
  async getByIdProxy(id: string | number): Promise<PlusApiResultPlusChannelProxyVO> {
    return this.client.get<PlusApiResultPlusChannelProxyVO>(backendApiPath(`/channel/proxy/${id}`));
  }

/** Delete channel proxy */
  async deleteProxy(id: string | number): Promise<PlusApiResultBoolean> {
    return this.client.delete<PlusApiResultBoolean>(backendApiPath(`/channel/proxy/${id}`));
  }

/** Get channel account by ID */
  async getByIdAccount(id: string | number): Promise<PlusApiResultPlusChannelAccountVO> {
    return this.client.get<PlusApiResultPlusChannelAccountVO>(backendApiPath(`/channel/account/${id}`));
  }

/** Delete channel account */
  async deleteAccount(id: string | number): Promise<PlusApiResultBoolean> {
    return this.client.delete<PlusApiResultBoolean>(backendApiPath(`/channel/account/${id}`));
  }
}

export function createChannelApi(client: HttpClient): ChannelApi {
  return new ChannelApi(client);
}
